base_layer/core/src/mempool/sync_protocol/mod.rs

```diff
@@ -174,6 +182,55 @@ where TSubstream: AsyncRead + AsyncWrite + Unpin + Send + Sync + 'static
         }
     }
 
+    async fn spawn_handle_block_event(&mut self, block_event: &BlockEvent) {
+        use BlockEvent::{BlockSyncComplete, ValidBlockAdded};
+        if self.permits.available_permits() < 1 {
+            // Sync is already in progress, so we should not bother trying to sync.
+            return;
+        }
+        match block_event {
+            ValidBlockAdded(_, BlockAddResult::ChainReorg { added, removed: _ }) => {
+                if added.len() < self.config.block_sync_trigger {
+                    return;
+                }
+            },
+            BlockSyncComplete(tip, starting_sync_height) => {
+                let added = tip.height() - starting_sync_height;
+                if added < self.config.block_sync_trigger as u64 {
+                    return;
+                }
+            },
+            _ => {
+                return;
+            },
+        }
+        // we need to make sure the service can start a sync
+        if self.num_synched.load(Ordering::SeqCst) >= self.config.initial_sync_num_peers {
+            self.num_synched.fetch_sub(1, Ordering::SeqCst);
+        }
```

Comment:
nit: slightly better use of atomic ordering
```suggestion
        if self.num_synched.load(Ordering::Aquire) >= self.config.initial_sync_num_peers {
            self.num_synched.fetch_sub(1, Ordering::Release);
        }
```